third-party/gmp/Makefile

```diff
@@ -11,11 +11,24 @@ include $(CHPL_MAKE_HOME)/make/Makefile.base
 GMP_UNPACKED_DIR = gmp-$(GMP_VERSION)
 GMP_TARBALL = $(GMP_UNPACKED_DIR)a.tar.bz2
 
+#
+# Cray X* builds are cross-compilations, as are Intel Phi KNC builds.
+#
+GMP_CROSS_COMPILED=no
+ifneq (, $(filter cray-x%,$(CHPL_MAKE_TARGET_PLATFORM)))
+CHPL_GMP_CFG_OPTIONS += --build=x86_64-unknown-linux-gnu \
+                        --host=x86_64-cnl-linux-gnu
+GMP_CROSS_COMPILED=yes
+else ifeq ($(CHPL_MAKE_TARGET_ARCH),knc)
+CHPL_GMP_CFG_OPTIONS += --host=x86_64-unknown-linux-gnu
```

Comment:
And now one more update.
With yet more studying of the GNU autoconf web page, I believe the right thing to do is to throw only --host, and not --build. In GNU autoconf 2.50 and beyond, cross compilation is enabled iff --host is thrown. All our third-party packages that use autoconf use versions >= 2.61, so we should be safe leaving off the --build option. I'm testing this now.